ApollosProject/apollos-prototype
# Adds Rock Request logging extension to apollo-datasource-rock

## DESCRIPTION

### What does this PR do, or why is it needed?
Adds rock request logging GraphQL Extension. 

- Uses two new class fields on RockApolloDataSource to collect data on number of rock requests made
- Exports a Class which will log data about the requests made to Rock.

### What design trade-offs/decisions were made?

- Chose to leave extension disabled by default. It logs a LOT of data for certain requests.
- Another option is to have this option be configurable via the `config.yml`. That requires a server restart to take effect though, and commenting the line out and back in doesn't 🤷‍♂️

### How do I test this PR?

1. `yarn build` && `yarn start`
2. Uncomment the commented out `import ...` statement and `extensions:..` statement in `src/server.js`
3. Click around in the app. You should see data logged in the API console about requests being made. 


### What automated tests did you write?


## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
